### PR TITLE
Refactor for package work

### PR DESCRIPTION
**Pull request by dlmr**
**Some of the things that are changed**
- Changes mentions of extensions to packages
- Adds logic to handle missing settings when printing
- Options/Arguments can now have a default value
- Options/Arguments can now define a convertor function that will transform the input to the correct format.
- Added start for convertors, better structure and can be used with arguments + options
- General cleanup/refactor
- Bug fixes
- Added runCli that can be used to more easily create a cli inside a package
- Updated the tests
- Updated the documentation  

---
- Added loud-rejection to the roc-cli and the ones using initCli
- Renamed debug to verbose
- Added hook(and actions) functionality
- Changed how roc.config.js functions
- Improved feedback messages from Roc.
- Changed how Roc resolves dependencies, now uses packages and plugins
- Added possibility to hide commands from markdownCommands and a title
- Removed a extra newline in the output
- Improved how the global data is managed
- Renamed important to info in helpers/style
- Changed how labels work in helpers/style
- Added a new helper function to create feedback messages
- Made roc itself to a Roc extension
- Updated tests to work with the changes in the code

### CODE REVIEW

**Comment by dlmr** [comment]
This is basically done at this time.

Some things that are missing:
- Test coverage on the new hook system that has been added.
- Update the code/documentation related to the existing templates when they are updated.

However this could come in a later PR.

File: src/commands/markdown-commands.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function markdownCommands(config, metaConfig, { name }, args, opt
 
         rows.push('## Commands');
         commands.forEach((command) => {
-            rows.push(`* [${command}](#${command})`);
+            rows.push(`* [${command}](#${command.replace(':', '')})`);
```
Comment (dlmr) [inline src/commands/markdown-commands.js:82]: Will probably need something a bit more robust here.

See https://gist.github.com/asabaylus/3071099

File: convertors/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/**
+* This file makes it easier to import the convertors.
+* Can use "roc/convertors" to only get the convertors.
+*/
+module.exports = require('../lib/convertors/index.js');
```
Comment (andreasrs) [inline convertors/index.js]: We should use the correct spelling converter + converters

File: src/cli/helpers.js [hunk 3]
```diff
@@ -557,7 +557,7 @@ export function getMappings(documentationObject = []) {
                 mappings[element.cli.substr(2)] = {
                     name: element.cli,
                     path: element.path,
-                    convertor: getConvertor(element.defaultValue, element.cli),
+                    converter: getConvertor(element.defaultValue, element.cli),
```
Comment (dlmr) [inline src/cli/helpers.js]: Not change the function name here also?

**Pull request merged**